### PR TITLE
Trying to enable thread fuzzer in flaky check

### PR DESCRIPTION
**Pull request by alesapin**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Build/Testing/Packaging Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Enable Thread Fuzzer for stateless tests flaky check.

### CODE REVIEW

File: src/Common/ThreadFuzzer.cpp [hunk 1]
```diff
@@ -102,6 +103,10 @@ void ThreadFuzzer::initConfiguration()
     initFromEnv(sleep_probability, "THREAD_FUZZER_SLEEP_PROBABILITY");
     initFromEnv(sleep_time_us, "THREAD_FUZZER_SLEEP_TIME_US");
 
+    LOG_DEBUG(&Poco::Logger::get("ThreadFuzzer"), "Sleep probability {}", sleep_probability);
```
Comment (alesapin) [inline src/Common/ThreadFuzzer.cpp:106]: Of course I'm going to remove this, just checking that env variables will be forwarded correctly.

**Comment by alesapin** [comment]
```
2020.12.21 15:15:46.031783 [ 141 ] {} <Debug> ThreadFuzzer: Sleep probability 0.1
2020.12.21 15:15:46.031812 [ 141 ] {} <Debug> ThreadFuzzer: Cpu time period 1000
2020.12.21 15:15:46.031833 [ 141 ] {} <Debug> ThreadFuzzer: Sleep time us 100000.0
```
Works

File: docker/test/stateless/run.sh [hunk 2]
```diff
@@ -12,7 +12,17 @@ dpkg -i package_folder/clickhouse-test_*.deb
 # install test configs
 /usr/share/clickhouse-test/config/install.sh
 
-service clickhouse-server start && sleep 5
+# For flaky check we also enable thread fuzzer
+if [ "$NUM_TRIES" -gt "1" ]; then
+    export THREAD_FUZZER_CPU_TIME_PERIOD_US=1000
+    export THREAD_FUZZER_SLEEP_PROBABILITY=0.1
+    export THREAD_FUZZER_SLEEP_TIME_US=100000
```
Comment (alexey-milovidov) [inline docker/test/stateless/run.sh:19]: Let's also wrap mutex ops.

Comment (alexey-milovidov) [reply docker/test/stateless/run.sh:19]: #9813, #9814

**Comment by alesapin** [comment]
```
2020.12.22 00:49:54.945305 [ 142 ] {} <Debug> ThreadFuzzer: Sleep probability 0.1
2020.12.22 00:49:54.945333 [ 142 ] {} <Debug> ThreadFuzzer: Yield probability 0.0
2020.12.22 00:49:54.945365 [ 142 ] {} <Debug> ThreadFuzzer: Migrate probability 0.0
2020.12.22 00:49:54.945396 [ 142 ] {} <Debug> ThreadFuzzer: CPU time period 1000
2020.12.22 00:49:54.945414 [ 142 ] {} <Debug> ThreadFuzzer: Sleep time us 100000.0
2020.12.22 00:49:54.946362 [ 142 ] {} <Debug> ThreadFuzzer: pthread_mutex_lock_before_migrate_probability 1.0
```
Works, removing logs.

**Comment by alesapin** [comment]
Lol, Fuzzer Works and the test really flaky!

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alesapin** [comment]
Yesss https://clickhouse-test-reports.s3.yandex.net/18299/d1d139a168ddeb2dd4d9a2af24fdb33bf58e6bec/functional_stateless_tests_flaky_check_(address).html

**Comment by alesapin** [comment]
Waiting for tests

**Comment by alesapin** [comment]
Manually cherry-picked fix for background remove of empty into 20.12 https://github.com/ClickHouse/ClickHouse/pull/17292

**Pull request merged**